node/reqres.js

```diff
@@ -285,21 +287,22 @@ TChannelOutgoingResponse.prototype.sendCallResponseContFrame = function sendCall
         case States.Initial:
             throw new Error('first response frame not sent'); // TODO: typed error
         case States.Streaming:
-            self.sendFrame.callResponseCont(isLast, args);
+            self.sendFrame.callResponseCont(args, isLast);
             if (isLast) self.state = States.Done;
             break;
         case States.Done:
+        case States.Error:
             throw new Error('response already done'); // TODO: typed error
     }
 };
 
 TChannelOutgoingResponse.prototype.sendErrorFrame = function sendErrorFrame(codeString, message) {
     var self = this;
-    if (self.state === States.Done) {
+    if (self.state === States.Done || self.state === States.Error) {
         throw new Error('response already done'); // TODO: typed error
     } else {
         self.sendFrame.error(codeString, message);
-        self.state = States.Done;
```

Comment:
Yes you need to be able to fast terminate a response with an Error frame at any time.